Query and cache ENS avatars (#5790)
rotki/rotki: docs/api.rst

   :statuscode 409: No user is currently logged in.
   :statuscode 500: Internal rotki error.
   :statuscode 502: An external service used in the query such as binance could not be reached or returned unexpected response.

Get avatar for an ens name
============================

.. http:get:: /api/(version)/avatars/ens/<ens_name>

   Doing a GET on this endpoint will return the avatar that is currently set for the provided ens name.

   **Example Request**

    .. http:example:: curl wget httpie python-requests

      POST /api/1/avatars/ens/rotki.eth HTTP/1.1
      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {}

      :reqjson string ens_name: The ens name to check avatar for. It should have `.eth` postfix.

      *Example Response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Content-Type: image/png

      :statuscode 200: The avatar was successfully found and returned.
      :statuscode 404: There is no avatar set for the given ens name.

>       :statuscode 409: Avatar was requested for an ens name that is not currently in the database.

Reviewer: yeah, you did. Thanks

   :statuscode 409: No user is currently logged in.
   :statuscode 500: Internal rotki error.
   :statuscode 502: An external service used in the query such as binance could not be reached or returned unexpected response.

Get avatar for an ens name
============================

.. http:get:: /api/(version)/avatars/ens/<ens_name>

>    Doing a GET on this endpoint will return the avatar that is currently set for the provided ens name. For successfull responses returns `ETag` header and for requests accepts `If-Match` and `If-None-Match` headers to allow caching on the client side.

Reviewer: ```suggestion
   Doing a GET on this endpoint will return the avatar that is currently set for the provided ens name. If successful, responses include an `ETag` header for caching.
```
